src/routes/DigitalAssets/routes/Send/sagas/digitalAssetsSend.js

```diff
@@ -89,116 +88,209 @@ function* openView(action: ExtractReturn<typeof digitalAssetsSend.openView>): Sa
   }
 }
 
-function* getAmountError(amount: string, digitalAsset: ?DigitalAsset): Saga<?string> {
-  const isValidAmount: boolean = (parseFloat(amount) > 0)
+function* requestGasPrice(): Saga<?BigNumber> {
+  const network: ExtractReturn<typeof selectCurrentNetwork> = yield select(selectCurrentNetwork)
 
-  if (!isValidAmount) {
-    return 'Amount should be greater than 0'
+  if (!network) {
+    throw new Error('There is no active network')
   }
 
-  if (!digitalAsset) {
+  try {
+    const gasPrice: BigNumber = yield call(web3.getGasPrice, network)
+    return gasPrice
+  } catch (err) {
     return null
   }
+}
 
+function* checkPriority(
+  digitalAsset: DigitalAsset,
+  priority: TXPriorityKey,
+  formFieldValues: DigitalAssetsSendFormFields,
+): Saga<boolean> {
   const {
-    address,
-    decimals,
-  }: DigitalAssetBlockchainParams = digitalAsset.blockchainParams
+    gasPrice: userGasPrice,
+    gasLimit: userGasLimit,
+  } = formFieldValues
+
+  if (priority === 'CUSTOM') {
+    const isGasPriceValid: boolean = parseFloat(userGasPrice) > 0
+    const isGasLimitValid: boolean = parseInt(userGasLimit, 10) > 0
+
+    if (!isGasLimitValid) {
+      yield put(digitalAssetsSend.setFormFieldError('gasLimit', 'Invalid value for gas limit'))
+    }
+
+    if (!isGasPriceValid) {
+      yield put(digitalAssetsSend.setFormFieldError('gasPrice', 'Invalid value for gas price'))
+    }
+
+    if (isGasLimitValid && isGasPriceValid) {
+      // convert it to WEI
+      const gasPrice = getTransactionValue(userGasPrice, 18).toString()
+      yield put(digitalAssetsSend.setGasPriceValue(gasPrice))
+
+      const gasLimit = toBigNumber(userGasLimit).toString()
+      yield put(digitalAssetsSend.setGasLimitValue(gasLimit))
```

Comment:
here setGasPriceValue/setGasLimitValue will rewrite default values, but they should be used to compare with user ones